src/plugins/plot/configuration/PlotSeries.js

```diff
@@ -394,7 +394,8 @@ export default class PlotSeries extends Model {
         const currentYVal = this.getYVal(point);
         const lastYVal = this.getYVal(data[insertIndex - 1]);
 
-        if (this.isValueInvalid(currentYVal) && this.isValueInvalid(lastYVal)) {
+        if (this.isValueInfinity(currentYVal)
```

Comment:
Based on order of operations this if statement will evaluate to true if one of these two cases is true:

* The currentYVal and lastYVal are invalid (including infinity)
OR 
* Just the currentYVal is infinity.

Why do we have this special case where the currentYVal is infinity? Why don't we also do this for other unplottable values?

Reply:
Honestly, this logic has been around for a while - I assumed it had something to do with data continuity. I can change it to fail for all unplottable values.

Reply:
The infinity logic is new though? We have a new OR statement which will shortcut the previous logic. Just wondering why?